odl/tomo/operators/xray_trafo.py

```diff
@@ -107,35 +112,44 @@ def __init__(self, discr_dom, geometry, backend='astra', **kwargs):
                 raise ValueError('ASTRA backend not available.')
             if not ASTRA_CUDA_AVAILABLE and self.backend == 'astra_cuda':
                 raise ValueError('ASTRA CUDA backend not available.')
-            if discr_dom.dspace.dtype not in (np.float32, np.complex64):
+            if discr_domain.dspace.dtype not in (np.float32, np.complex64):
                 raise ValueError('ASTRA support is limited to `float32` for '
                                  'real and `complex64` for complex data.')
-            if not np.allclose(discr_dom.grid.stride[1:],
-                               discr_dom.grid.stride[:-1]):
+            if not np.allclose(discr_domain.grid.stride[1:],
+                               discr_domain.grid.stride[:-1]):
                 raise ValueError('ASTRA does not support different pixel/voxel'
                                  ' sizes per axis (got {}).'
-                                 ''.format(discr_dom.grid.stride))
+                                 ''.format(discr_domain.grid.stride))
 
         self._geometry = geometry
 
-        # Create discretized space (operator range). Use same data space
+        # Create a discretized space (operator range) with the same data space
         # type as the domain.
         # TODO: maybe use a ProductSpace structure
-        ran_uspace = FunctionSpace(geometry.params)
-        # CHECKME: Is this the right weight?
+        range_uspace = FunctionSpace(geometry.params)
 
         weight = getattr(geometry.grid, 'cell_volume', 1.0)
-
-        ran_dspace = discr_dom.dspace_type(geometry.grid.ntotal,
-                                           weight=weight,
-                                           dtype=discr_dom.dspace.dtype)
-
-        ran_interp = kwargs.pop('range_interpolation', 'nearest')
-        discr_ran = DiscreteLp(ran_uspace, geometry.grid, ran_dspace,
-                               interp=ran_interp, order=geometry.grid.order)
-        super().__init__(discr_dom, discr_ran, linear=True)
-
-        self._adjoint = DiscreteXrayTransformAdjoint(self)
+        if isinstance(geometry, HelicalConeFlatGeometry):
```

Comment:
We need to go back and look into the stuff I wrote up. The weighting in this place is only with respect to the **parametrization**, i.e. if we happen to choose the parameter rectangle to coincide with the size of the (flat) detector, we get weight equal to the cell size in that parameter space. The exact same is true for curved detectors with the only difference that one or two of the parameters are angles - but they still lie on a rectangle. The shape of the detector is handled by the weight function defined in `Detector` (`surface_measure` I think it is). Finally, geometric relations like acquisition curve, radii etc. are handled by the back projector.

So, in conclusion: push this geometry related weighting to the back projector and (for later) weightings due to curvature to the `surface_measure` of the detector.